File: 6. Zigzag Conversion/zigzag.kt
class Solution{

    fun convert(s: String, numRows: Int): String{

        if(numrows == 1 || s.length<=numRows)return s

        val rows=Array(numRows) { StringBuilder() }

        var direction=1
        var currentRow=0

        for(c in s){

            rows[currentRow].append(c)

            if(curentRow == 0)direction=1
            else if(currentRow==numRows -1) direction =-1

            currentRow+=direction
        }

        val result = StringBuilder()

        for(row in rows){
            result.append(row)
        }

        return result.toString()

    }
}

fun main(){


    val s=readln()

    val numRows=readln().toInt()

    val obj = Solution()

    println(obj.convert(s,numRows))
}

File: 6. Zigzag Conversion/zigzag.cpp
#include <bits/stdc++.h>
using namespace std;

class  Solution{
public:
        string convert(string s, int numRows){
            if(numRows==1 || s.size()<=numRows) return s;

            vector<string> rows(numRows);

            int currentRow=0;
            int direction=1;

            for(char c: s){

                row[currentRow]+=c;

                if(currentRow ==0) direction=1;
                else if(direction ==numRow -1) direction=-1;

                currentRow+=direction;
            }

            string result="";

            for(string row: rows){
                result+=row;
            }

            return result;

        }

};

int main() {

    string s;
    int numRows;

    cin >> s;
    cin >> numRows;

    Solution obj;

    cout << obj.convert(s, numRows);

}
